### PR TITLE
Restructure code for issue #14.

### PR DESCRIPTION
**Pull request by gburtini**
This is a lot of changes that build on top of #16 and #17. If you merge them first you won't see the duplicated changes.

This restructures the code in two primary ways:
1. To clearly support `node` usage without the browserify compile step.
2. To distinguish between distribution files and source files.
3. To separate the `npm` scripts in to chunks. `build` still exists, but it just runs `jison` and `dist` now.
4. To add a [yarn.lock](https://yarnpkg.com/blog/2016/11/24/lockfiles-for-all/).
5. To standardize case on `camelCase` where appropriate.

I've run the tests, it still works (hopefully Travis CI says the same!). There's a few more changes I want to make but they didn't belong in this PR.

Upon merge, this should close issue #14.

### CODE REVIEW

**Comment by gburtini** [comment]
Travis failed, I am investigating why.

**Comment by gburtini** [comment]
It is because of bad filename case handling. I am on it.

File: yarn.lock [hunk 1]
```diff
@@ -0,0 +1,1518 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
+# yarn lockfile v1
+
+
+JSONSelect@0.4.0:
```
Comment (emord) [inline yarn.lock]: Where are these coming from? I don't see a `JSONSelcect` as a dependency locally

Comment (gburtini) [inline yarn.lock]: Yarn locks the version of all dependencies, including subdependencies. This particular dependency comes from Browserify. Think of `yarn.lock` like `npm-shrinkwrap.json`.

**Comment by gburtini** [comment]
Flattened my commit history as requested.

File: yarn.lock (hunk 1 above)
Comment (emord) [inline yarn.lock]: Thanks. I missed that when looking at browserfiy's dependencies

File: README.md [hunk 2]
```diff
@@ -47,21 +47,22 @@ Known Limitations
...
 
-| File      | Content                                            |
-| ---------:|:---------------------------------------------------|
-| xpath.js  | the generated parser file.                         |
-| models.js | the underlying models files used in parsing.       |
-| src/      | contains the jison parser files.                   |
-| lib/      | external libraries that the parser depends on.     |
-| test/     | qunit tests for the parser and generator.          |
+| File           | Content                                            |
+| --------------:|:---------------------------------------------------|
```
Comment (millerdev) [inline README.md]: Guessing this alignment decision was intentional, although I find it hard to read once the markdown is converted to HTML. PR here https://github.com/dimagi/js-xpath/pull/26, but feedback is welcome.

Comment (gburtini) [inline README.md]: Nope, totally wasn't intentional. I copied a Markdown table example. :)

**Pull request merged**